text/profiles/0239-profiles-data-model.md

```diff
@@ -0,0 +1,1601 @@
+
+* Sharing of the call stacks between samples.
+* Sharing of labels (now called attributes) between samples.
+* Reuse of OpenTelemetry conventions and message types.
+* Semantic conventions for linking to other signals via `trace_id`s and `span_id`s.
+* First-class timestamp support.
+
+Below you will find the proto for the new Profiles signal. It is split into two parts: the first part is the OpenTelemetry specific part, and the second part is the modified pprof proto. Intention here is to make it easier to compare modified pprof proto to the original pprof proto.
+
+OpenTelemetry specific part:
+<!-- proto1 -->
+
+```proto
+syntax = "proto3";
+
+package opentelemetry.proto.profiles.v1;
+
+import "opentelemetry/proto/common/v1/common.proto";
+import "opentelemetry/proto/resource/v1/resource.proto";
+
+import "opentelemetry/proto/profiles/v1/alternatives/pprofextended/pprofextended.proto";
+
+option csharp_namespace = "OpenTelemetry.Proto.Profiles.V1";
+option java_multiple_files = true;
+option java_package = "io.opentelemetry.proto.profiles.v1";
+option java_outer_classname = "ProfilesProto";
+option go_package = "go.opentelemetry.io/proto/otlp/profiles/v1";
+
+//                Relationships Diagram
+//
+// ┌──────────────────┐                LEGEND
+// │   ProfilesData   │
+// └──────────────────┘            ─────▶ embedded
+//   │
+//   │ 1-n                         ─────▷ referenced by index
+//   ▼
+// ┌──────────────────┐
+// │ ResourceProfiles │
+// └──────────────────┘
+//   │
+//   │ 1-n
+//   ▼
+// ┌──────────────────┐
+// │  ScopeProfiles   │
+// └──────────────────┘
+//   │
+//   │ 1-n
+//   ▼
+// ┌──────────────────┐
+// │ ProfileContainer │
+// └──────────────────┘
+//   │
+//   │ 1-1
+//   ▼
+// ┌──────────────────┐
+// │      Profile     │
+// └──────────────────┘
+//   │                                1-n
+//   │ 1-n         ┌───────────────────────────────────────┐
+//   ▼             │                                       ▽
+// ┌──────────────────┐   1-n   ┌──────────────┐      ┌──────────┐
+// │      Sample      │ ──────▷ │   KeyValue   │      │   Link   │
+// └──────────────────┘         └──────────────┘      └──────────┘
+//   │                    1-n       △      △
+//   │ 1-n        ┌─────────────────┘      │ 1-n
+//   ▽            │                        │
+// ┌──────────────────┐   1-1   ┌──────────────┐
+// │     Location     │ ──────▷ │   Mapping    │
+// └──────────────────┘         └──────────────┘
+//   │
+//   │ 1-n
+//   ▼
+// ┌──────────────────┐
+// │       Line       │
+// └──────────────────┘
+//   │
+//   │ 1-1
+//   ▽
+// ┌──────────────────┐
+// │     Function     │
+// └──────────────────┘
+//
+
+// ProfilesData represents the profiles data that can be stored in a persistent storage,
```

Comment:
FWIW I think plural in noun phrases is uncommon, so ProfileData would be more common. But maybe this follows some common OTel convention.

Reply:
This is OTEL specific, going to go with convention:
* https://github.com/open-telemetry/opentelemetry-proto/blob/449a1b0219a699a60d9cbc3899c68cb1ec8c1139/opentelemetry/proto/metrics/v1/metrics.proto#L38
* https://github.com/open-telemetry/opentelemetry-proto/blob/449a1b0219a699a60d9cbc3899c68cb1ec8c1139/opentelemetry/proto/logs/v1/logs.proto#L38
* https://github.com/open-telemetry/opentelemetry-proto/blob/c26513af3305a845d1b38abea6244b074d21759d/opentelemetry/proto/trace/v1/trace.proto#L38